dashboard/src/components/entitypage/EntityPageView.js

```diff
@@ -226,9 +233,30 @@ const EntityPageView = ({ entity, setVersion, activeVersions }) => {
     setVersion(type, name, resources["default-variant"]);
   }
 
-  let resource = resources.versions[version];
-  const metadata = resource.metadata;
-  let resourceData = resource.data;
+  let resource;
+  if (resourceVersions[type]) {
+    resource = resources.versions[version];
+  } else {
+    resource = resources;
+  }
+  let metadata = {};
+  let resourceData = {};
+  if (local) {
```

Comment:
Why are these things different? Could we have two classses, a LocalImplementation and RESTImplemenation so that all the logic lives in one place and not across a handful of if(local) statements